File: escadinha/escadinhaLambda.js
function contarEscadinhas(N, sequencia) {
    if (N <= 2)
        return 1;  // Qualquer sequência com 1 ou 2 números é uma escadinha

    let diferencaAtual = sequencia[1] - sequencia[0];
    let quantidadeEscadinhas = 1;

    for (let valor = 2; valor < N; valor++) {
        if (sequencia[valor] - sequencia[valor - 1] !== diferencaAtual) {
            diferencaAtual = sequencia[valor] - sequencia[valor - 1];
            quantidadeEscadinhas++;
        }
    }

    return quantidadeEscadinhas;
}

const readline = require('readline');
// import readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Informe o tamanho da sequência: ', (N) => {
    rl.question('Informe a sequência de números separados por espaço: ', (sequenciaStr) => {
        const sequencia = sequenciaStr.split(' ').map(Number);
        const resultado = contarEscadinhas(N, sequencia);
        console.log(resultado);
        rl.close();
    });
});

File: escadinha/escadinha_lambda.py
N = int(input())
sequência = list(map(int, input().split()))

diferença_atual = sequência[1] - sequência[0]
quantidade_de_escadinhas_encontradas = 1

compara_diferenças = (
    lambda valor_atual, valor_anterior: valor_atual - valor_anterior == diferença_atual
)

for valor in range(2, N):
    if not compara_diferenças(sequência[valor], sequência[valor - 1]):
        diferença_atual = sequência[valor] - sequência[valor - 1]
        quantidade_de_escadinhas_encontradas += 1
